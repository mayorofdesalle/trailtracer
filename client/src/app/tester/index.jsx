import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
