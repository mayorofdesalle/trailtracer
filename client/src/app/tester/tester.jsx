import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };