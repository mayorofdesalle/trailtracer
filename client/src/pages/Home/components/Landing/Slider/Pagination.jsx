import { memo } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Button from '@components/ui/Button';
import { fadeIn } from '@components/misc/anims';

const pages = [{ name: 'Home', id: 0 }, { name: 'Features', id: 1 }, { name: 'Features', id: 2 }, { name: 'Features', id: 3 }];

/**
 * PageDot
 * @description
 * This is a styled container that is used to create a page dot.
 **/
const PageDot = styled(Button)`
    padding: 0;
    height: 0.75svh;
    width: 0.75svh;
    border-radius: 0.125rem;
    transform: rotate(45deg);
    background-color: ${({ theme }) => theme.colors.textTransparent};
    animation: ${fadeIn} 1s ease-in-out;

    // To incease the hover area
    &::before {
        content: '';
        position: absolute;
        border-radius: 50%;
        width: 500%;
        height: 500%;
        left: 50%;
        top: 50%;
        transform: translate(-50%, -50%);
        z-index:-1;
    }

    &:hover {
        background-color: ${({ theme }) => theme.colors.text};
        transform: rotate(90deg) scale(1.5);
    }

    &:active {
        transform: rotate(135deg) scale(1);
    }
`;

/**
 * Pagination
 * @description
 * This is a function component that is used to contain all the page dots.
 **/
const Pagination = memo(({ onClick }) => {
    return pages.map((page) => (
        <PageDot onClick={() => onClick(page.id)} key={page.id} />
    ));
});

Pagination.propTypes = {
    onClick: PropTypes.func,
};

export default Pagination;