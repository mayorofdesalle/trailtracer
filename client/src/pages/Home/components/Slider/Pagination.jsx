import styled from 'styled-components';

import { glassMorph } from '@components/misc/mixins';
import { rotateInFB } from '@components/misc/anims';

/**
 * Pagination
 * @description
 * This is a styled unordered list that is used to contain the page dots.
 **/
const Pagination = styled.ul`
    grid-area: Slider;
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: space-around;
    align-items: center;
    outline: 2px solid ${({ theme }) => theme.colors.primaryTransparent};
    border-radius: 10px;
    animation: ${rotateInFB} 1s ease-in-out;

    & > .isDragging {
        background-color: ${({ theme }) => theme.colors.primary};
        transition: none;
    }

    & > .isDragging > * {
        fill: ${({ theme }) => theme.colors.backgroundTransparent};
    }

    ${glassMorph}
`;

export default Pagination;