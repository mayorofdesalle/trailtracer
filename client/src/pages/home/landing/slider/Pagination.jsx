import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Button } from '@components/ui';

const pages = [{ name: 'Home', id: 0 }, { name: 'Features', id: 1 }, { name: 'Features', id: 2 }, { name: 'Features', id: 3 }];

const PageDot = styled(Button)`
    position: relative;
    padding: 0;
    height: 1dvh;
    width: 0.75dvh;
    border-radius: max(0.125dvh, 0.125dvw, 0.125rem);
    background-color: ${({ theme }) => theme.colors.textTransparent};

    // To incease the hover area
    &::before {
        content: '';
        position: absolute;
        border-radius: 50%;
        border: 1px solid ${({ theme }) => theme.colors.primary};
        width: 500%;
        height: 500%;
        left: 50%;
        top: 50%;
        transform: translate(-50%, -50%);
        z-index: -1;
        opacity: 0;
    }

    &:hover {
        border-radius: max(0.0625dvh, 0.0625dvw, 0.0625rem);
        background-color: ${({ theme }) => theme.colors.text};
        transform: scale(1.5);
    }

    &:active {
        border-radius: max(0.125dvh, 0.125dvw, 0.125rem);
        transform: scale(1);
    }
`;

/**
 * Pagination
 * @description
 * Dots that represent the pages in the slider.
 **/
const Pagination = ({ onClick }) => {
    return pages.map((page) => (
        <PageDot onClick={() => onClick(page.id)} key={page.id} $animDelay={page.id} />
    ));
};

Pagination.propTypes = {
    onClick: PropTypes.func,
};

export default Pagination;