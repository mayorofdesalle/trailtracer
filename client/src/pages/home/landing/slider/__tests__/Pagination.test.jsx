import { describe, expect, it, render, screen } from '@app/tester';

import Pagination from '../Pagination';

describe('Pagination', () => {

    it('should render correctly', () => {
        render(<Pagination />);
        const pageDots = screen.getAllByRole('button');
        expect(pageDots).toHaveLength(4);
        pageDots.forEach(dot => expect(dot).toBeInTheDocument());
    });

});