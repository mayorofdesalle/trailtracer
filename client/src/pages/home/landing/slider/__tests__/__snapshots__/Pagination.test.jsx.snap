// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-beySPh sc-irLvIq zbkUW kNKvsb"
    type="button"
  />
  <button
    class="sc-beySPh sc-irLvIq zbkUW kNKvsb"
    type="button"
  />
  <button
    class="sc-beySPh sc-irLvIq zbkUW kNKvsb"
    type="button"
  />
  <button
    class="sc-beySPh sc-irLvIq zbkUW kNKvsb"
    type="button"
  />
</DocumentFragment>
`;
